core/domain/learner_progress_services_test.py

```diff
@@ -481,6 +485,43 @@ def test_get_all_completed_exp_ids(self):
         self.assertEqual(learner_progress_services.get_all_completed_exp_ids(
             self.user_id), [self.EXP_ID_0, self.EXP_ID_1])
 
+    def test_get_filtered_completed_exp_summaries(self):
+        # Add explorations to the completed list.
+        learner_progress_services.mark_exploration_as_completed(
+            self.user_id, self.EXP_ID_0)
+        learner_progress_services.mark_exploration_as_completed(
+            self.user_id, self.EXP_ID_1)
+
+        # Mark EXP_ID_2 as completed and create summary list to test output.
+        learner_progress_services.mark_exploration_as_completed(
+            self.user_id, self.EXP_ID_2)
+        exploration_summaries = (
+            exp_services.get_exploration_summaries_matching_ids(
+                learner_progress_services.get_all_completed_exp_ids(
+                    self.user_id)))
+
+        # Unpublish EXP_ID_2 to change status to ACTIVITY_STATUS_PRIVATE.
```

Comment:
Can we assert that initially there were three exploration_ids in the list? (as a baseline verification.)

Ditto elsewhere!

Reply:
Good point. Done!